docs/sources/flow/getting-started/controllers.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Kuberenetes Deployment Modes
+---
+## StatefulSet
+
+A `StatefulSet` is the most common way Grafana Agent is deployed in Kubernetes. Advantages include:
+
+- A relatively small number of Agents can handle metrics for an entire cluster.
+- Persistent volumes can be attatched to each pod to persist agent data like the write-ahead log or other component state.
+- Number of pods can be scaled up or down as needed, either manually or with horizontal pod autoscaling.
+
+### Distributing work with StatefulSets
+
+Grafana Agent's [built-in clustering](https://grafana.com/docs/agent/latest/flow/getting-started/distribute-prometheus-scrape-load/) is the easiest way to distribute scrape load across a `StatefulSet` of Agents. 
+
+## Deployment
+
+A `Deployment` has most of the same properties as a `StatefulSet`, but cannot use Persistent Volume Claim Templates. It may be desired if you do not want storage, but most of the other properties are the same as a `StatefulSet`.
+
+## DaemonSet
+
+A `DaemonSet` will deploy a Grafana Agent pod to each Node in your Kubernetes cluster. This is needed when you need access to something that is present on every Node in your cluster, such as:
+
+- Local access to log files.
+- Exporters, such as `node_exporter`, which run on every physical node.
+
+There are several downsides to a `DaemonSet` which need to be considered:
+
+- You cannot attach Persistent Volumes to `DaemonSet` pods. This means things like the Write-Ahead Log will use the Node's local disk, which may not be desired in shared environments.
+
+- Each Agent Pod with remote write components will have its' own Write-Ahead Log. This could increase the total memory used across all agents, especially if there is a temporary disruption writing to the backend storage.
+
+### Distributing work in a DaemonSet
+
+// TODO: should this go in the page on "distributing prometheus scrape"?
+
+In order to avoid scraping targets multiple times, it is best to have each Grafana Agent in a DaemonSet scrape targets only on its' local Node. The most performant way to do this is to apply a field selector to your `discovery.kubernetes` components that discover pods or endpoints:
+
+```river
+discovery.kubernetes "pods" {
+    role = "pod"
+    selectors {
+        role = "pod"
+        field = "spec.nodeName=" + constants.hostname
+    }
+}
+discovery.kubernetes "endpoints" {
+    role = "endpoint"
+    selectors {
+        role = "pod"
```

Comment:
This is already documented in the reference docs, so not sure if it's worth calling out again here, but I imagine the two `role` arguments could be a bit confusing to newer folks?

Maybe a sentence like:
```markdown
The top level `role` argument specifies the type of Kubernetes resource to query, and the selectors' `role` specifies the role of the selector.
```